Libraries/PeriphDrivers/Source/I3C/i3c_reva.c

```diff
@@ -612,7 +616,9 @@ int MXC_I3C_RevA_Controller_Transaction(mxc_i3c_reva_regs_t *i3c, const mxc_i3c_
             goto err;
         }
 
-        ret = MXC_I3C_RevA_WriteTXFIFO(i3c, req->tx_buf, req->tx_len, true, 100);
+        timeout = (uint32_t)(40 * 1000000 / freq) * req->tx_len;
```

Comment:
Would be helpful to add a comment on what the formula is or how this timeout calculation was derived.